diff -Nru linux-3.7-rc5.orig/drivers/gpu/drm/i915/i915_suspend.c linux-3.7-rc5/drivers/gpu/drm/i915/i915_suspend.c
--- linux-3.7-rc5.orig/drivers/gpu/drm/i915/i915_suspend.c	2012-10-21 18:32:56.000000000 +0200
+++ linux-3.7-rc5/drivers/gpu/drm/i915/i915_suspend.c	2012-10-22 23:56:31.000000000 +0200
@@ -877,6 +877,8 @@
 		I915_WRITE(IMR, dev_priv->saveIMR);
 	}
 
+	msleep (10);
+
 	/* Cache mode state */
	I915_WRITE(CACHE_MODE_0, dev_priv->saveCACHE_MODE_0 | 0xffff0000);

diff -Nru linux-3.7-rc5.orig/drivers/gpu/drm/i915/intel_pm.c linux-3.7-rc5/drivers/gpu/drm/i915/intel_pm.c
--- linux-3.7-rc5.orig/drivers/gpu/drm/i915/intel_pm.c	2012-10-01 01:47:46.000000000 +0200
+++ linux-3.7-rc5/drivers/gpu/drm/i915/intel_pm.c	2012-11-26 16:54:48.340168250 +0100
@@ -3328,6 +3328,8 @@
 	I915_WRITE(WM2_LP_ILK, 0);
 	I915_WRITE(WM1_LP_ILK, 0);
 
+	msleep(10);
+
 	I915_WRITE(CACHE_MODE_0,
 		   _MASKED_BIT_DISABLE(CM0_STC_EVICT_DISABLE_LRA_SNB));
